vpr/src/place/place_util.h

```diff
@@ -5,33 +5,137 @@
+ * so we store their multiplicative inverses when they are updated in
+ * the outer loop routines to speed up the normalization process.
+ *
+ *   @param cost The weighted average of the wiring cost and the timing cost.
+ *   @param bb_cost The bounding box cost, aka the wiring cost.
+ *   @param timing_cost The timing cost, which is connection delay * criticality.
+ *
+ *   @param bb_cost_norm The normalization factor for the wiring cost.
+ *   @param timing_cost_norm The normalization factor for the timing cost, which
+ *              is upper-bounded by the value of MAX_INV_TIMING_COST.
+ *
+ *   @param MAX_INV_TIMING_COST Stops inverse timing cost from going to infinity
+ *              with very lax timing constraints, which avoids multiplying by a
+ *              gigantic timing_cost_norm when auto-normalizing. The exact value
+ *              of this cost has relatively little impact, but should not be large
+ *              enough to be on the order of timing costs for normal constraints.
+ *
+ *   @param place_algorithm Determines how the member values are updated upon
+ *              each temperature change during the placer annealing process.
+ */
+class t_placer_costs {
+  public: //members
     double cost;
     double bb_cost;
     double timing_cost;
-};
+    double bb_cost_norm;
+    double timing_cost_norm;
 
-struct t_placer_prev_inverse_costs {
-    double bb_cost;
-    double timing_cost;
+  public: //Constructor
+    t_placer_costs(t_place_algorithm algo)
+        : place_algorithm(algo) {}
+
+  public: //Mutator
+    void update_norm_factors();
+
+  private:
+    double MAX_INV_TIMING_COST = 1.e9;
+    t_place_algorithm place_algorithm;
 };
 
-// Used by update_annealing_state()
-struct t_annealing_state {
-    float t;                  // Temperature
-    float rlim;               // Range limit for swaps
-    float inverse_delta_rlim; // used to calculate crit_exponent
-    float alpha;              // Temperature decays by this factor each outer iteration
-    float restart_t;          // Temperature used after restart due to minimum success ratio
-    float crit_exponent;      // Used by timing-driven placement to "sharpen" timing criticality
-    int move_lim_max;         // Maximum move limit
-    int move_lim;             // Current move limit
+/**
+ * @brief Stores variables that are used by the annealing process.
+ *
+ * This structure is updated by update_annealing_state() on each outer
+ * loop iteration. It stores various important variables that need to
+ * be accessed during the placement inner loop.
+ *
+ * Public members:
+ *   @param t
+ *              Temperature for simulated annealing.
+ *   @param rlim
+ *              Range limit for block swaps.
+ *   @param alpha
+ *              Temperature decays factor (multiplied each outer loop iteration).
+ *   @param restart_t
+ *              Temperature used after restart due to minimum success ratio.
```

Comment:
I think this is only used with DUSTY_SCHED; please check and mention that here if I'm right.